iam/src/providers/githubForkedRepoProvider.ts

```diff
@@ -0,0 +1,131 @@
+// ----- Types
+import type { RequestPayload, VerifiedPayload } from "@gitcoin/passport-types";
+import type { Provider, ProviderOptions } from "../types";
+import type {
+  GithubFindMyUserResponse,
+  GithubUserRepoResponseData,
+  GithubRepoRequestResponse,
+} from "./types/githubTypes";
+
+// ----- HTTP Client
+import axios from "axios";
+
+export type GithubTokenResponse = {
+  access_token: string;
+};
+
+// Export a Github Provider to carry out OAuth and return a record object
+export class ForkedGithubRepoProvider implements Provider {
+  // Give the provider a type so that we can select it with a payload
+  type = "ForkedGithubRepoProvider";
+
+  // Options can be set here and/or via the constructor
+  _options = {};
+
+  // construct the provider instance with supplied options
+  constructor(options: ProviderOptions = {}) {
+    this._options = { ...this._options, ...options };
+  }
+
+  // verify that the proof object contains valid === "true"
+  async verify(payload: RequestPayload): Promise<VerifiedPayload> {
+    let valid = false,
+      verifiedUserPayload: GithubFindMyUserResponse = {},
+      verifiedUserRepoPayload: GithubUserRepoResponseData = {};
+
+    try {
+      verifiedUserPayload = await verifyGithub(payload.proofs.code);
```

Comment:
I would recommend requesting the access token at the beginning of this block, and passing it as a parameter to the `verify*` function.
Then these functions do not need to request a token, and we save 1 request to github.